packages/backend/src/models/CatalogModel/CatalogModel.ts

```diff
@@ -107,7 +107,18 @@ export class CatalogModel {
             {
                 catalogSize: catalogItems.length,
             },
-            async () => catalogItems.map(parseCatalog),
+            async () =>
```

Comment:
if we're not planning on `showing` these fields, you could filter them out on `index` time.

Reply:
And by that, do you mean when we get the explore from `cached_explore.explore`?

I got a SQL query that works

```SQL
WITH updated_explore AS (
    SELECT
        table_names,
        jsonb_set(
            explore,
            '{tables}',
            (
                SELECT jsonb_object_agg(table_name, table_data)
                FROM (
                    SELECT
                        key AS table_name,
                        jsonb_set(
                            jsonb_set(
                                table_data,
                                '{metrics}',
                                (
                                    SELECT jsonb_object_agg(metric_key, metric_value)
                                    FROM jsonb_each(table_data->'metrics') AS metrics(metric_key, metric_value)
                                    WHERE (metric_value->>'hidden')::boolean IS NOT TRUE
                                )
                            ),
                            '{dimensions}',
                            (
                                SELECT jsonb_object_agg(dimension_key, dimension_value)
                                FROM jsonb_each(table_data->'dimensions') AS dimensions(dimension_key, dimension_value)
                                WHERE (dimension_value->>'hidden')::boolean IS NOT TRUE
                            )
                        ) AS table_data
                    FROM jsonb_each(explore->'tables') AS tables(key, table_data)
                ) AS filtered_tables
            )
        ) AS updated_explore
    FROM cached_explore
)
SELECT updated_explore
FROM updated_explore;

```

And was trying to do it with as many knex methods as I could, but I've seen they don't provide us with a `jsonb_each` equivalent. 

And I'm not sure if we can do nested jsonSet calls... What do you think @rephus ?

Reply:
Let's ignore this because what was meant https://github.com/lightdash/lightdash/pull/10490#discussion_r1652362088